public/app/features/dashboard-scene/settings/variables/utils.ts

```diff
@@ -0,0 +1,112 @@
+import { SelectableValue } from '@grafana/data';
+import {
+  AdHocFiltersVariable,
+  ConstantVariable,
+  CustomVariable,
+  DataSourceVariable,
+  IntervalVariable,
+  TextBoxVariable,
+  QueryVariable,
+} from '@grafana/scenes';
+import { VariableType } from '@grafana/schema';
+
+import { AdHocFiltersVariableEditor } from './editors/AdHocFiltersVariableEditor';
+import { ConstantVariableEditor } from './editors/ConstantVariableEditor';
+import { CustomVariableEditor } from './editors/CustomVariableEditor';
+import { DataSourceVariableEditor } from './editors/DataSourceVariableEditor';
+import { IntervalVariableEditor } from './editors/IntervalVariableEditor';
+import { QueryVariableEditor } from './editors/QueryVariableEditor';
+import { TextBoxVariableEditor } from './editors/TextBoxVariableEditor';
+
+interface EditableVariableConfig {
+  name: string;
+  description: string;
+  // FIXME: This should include all the scene objects
+  scene:
+    | typeof CustomVariable
+    | typeof QueryVariable
+    | typeof ConstantVariable
+    | typeof IntervalVariable
+    | typeof DataSourceVariable
+    | typeof AdHocFiltersVariable
+    | typeof TextBoxVariable;
```

Comment:
Can we maybe type this as `SceneVariable`, available from grafana/scenes? We already haev `sceneUtils.isXXXXVariable(variable: SceneVariable)` type guards that would allow us to do any type checks that we would require.

Reply:
I think it was not working, and I don't remember why. I can give it a second try.

Reply:
Refactored with the solution we agreed on!